src/main/java/org/opensearch/performanceanalyzer/collectors/telemetry/RTFDisksCollector.java

```diff
@@ -0,0 +1,113 @@
+    private static final Logger LOG = LogManager.getLogger(RTFDisksCollector.class);
+    private PerformanceAnalyzerController performanceAnalyzerController;
+    private ConfigOverridesWrapper configOverridesWrapper;
+
+    public RTFDisksCollector(
+            PerformanceAnalyzerController performanceAnalyzerController,
+            ConfigOverridesWrapper configOverridesWrapper) {
+        super(
+                MetricsConfiguration.CONFIG_MAP.get(RTFDisksCollector.class).samplingInterval,
+                "RTFDisksCollector",
+                StatMetrics.DISKS_COLLECTOR_EXECUTION_TIME,
+                StatExceptionCode.DISK_METRICS_COLLECTOR_ERROR);
+        this.metricsInitialised = false;
+        this.performanceAnalyzerController = performanceAnalyzerController;
+        this.configOverridesWrapper = configOverridesWrapper;
+    }
+
+    @Override
+    public void collectMetrics(long startTime) {
+        if (performanceAnalyzerController.isCollectorDisabled(
+                configOverridesWrapper, getCollectorName())) {
+            LOG.info("RTFDisksCollector is disabled. Skipping collection.");
+            return;
+        }
+
+        OSMetricsGenerator generator = OSMetricsGeneratorFactory.getInstance();
+        if (generator == null) {
+            LOG.error("could not get the instance of OSMetricsGeneratorFactory class");
+            return;
+        }
+
+        metricsRegistry = OpenSearchResources.INSTANCE.getMetricsRegistry();
+        if (metricsRegistry == null) {
+            LOG.error("could not get the instance of MetricsRegistry class");
+            return;
+        }
+
+        LOG.debug("Executing collect metrics for RTFDisksCollector");
+
+        initialiseMetricsIfNeeded();
+        DiskMetricsGenerator diskMetricsGenerator = generator.getDiskMetricsGenerator();
+        diskMetricsGenerator.addSample();
+
+        recordMetrics(diskMetricsGenerator);
+    }
+
+    public void recordMetrics(DiskMetricsGenerator diskMetricsGenerator) {
+        for (String disk : diskMetricsGenerator.getAllDisks()) {
+            Tags diskNameTag =
+                    Tags.create().addTag(RTFMetrics.DiskDimension.DISK_NAME.toString(), disk);
+            double diskWaitTime = diskMetricsGenerator.getAwait(disk);
+            double diskServiceRate = diskMetricsGenerator.getServiceRate(disk);
+            double diskUtilization = diskMetricsGenerator.getDiskUtilization(disk);
+            diskWaitTimeMetrics.record(diskWaitTime, diskNameTag);
+            diskUtilizationMetrics.record(diskUtilization, diskNameTag);
+            diskServiceRateMetrics.record(diskServiceRate, diskNameTag);
+        }
+    }
+
+    private void initialiseMetricsIfNeeded() {
+        if (metricsInitialised == false) {
+            diskWaitTimeMetrics =
+                    metricsRegistry.createHistogram(
+                            RTFMetrics.DiskValue.Constants.WAIT_VALUE,
+                            "DiskWaitTimeMetrics",
```

Comment:
Please follow the otel convention for metric names as well.

Reply:
That's metrics description. Metric name is the first parameter